src/components/screens/sprint-view-screen.tsx

```diff
@@ -1,13 +1,242 @@
+  const [projects, setProjects] = useState<Projects[]>([]);
+  const [timeEntries, setTimeEntries] = useState<number[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [myTasks, setMyTasks] = useState(true);
+  const [filter, setFilter] = useState("");
+  const todaysDate = new Date().toISOString();
+  const sprintStartDate = getSprintStart(todaysDate);
+  const sprintEndDate = getSprintEnd(todaysDate);
+  const columns = sprintViewProjectsColumns({ allocations, timeEntries, projects });
+  const setError = useSetAtom(errorAtom);
+
+  /**
+   * Get project data if user is logged in
+   */
+  useEffect(() => {
+    fetchProjectDetails();
+  }, [loggedInPerson]);
+
+  /**
+   * Fetch allocations, project names and time entries
+   */
+  const fetchProjectDetails = async () => {
+    setLoading(true);
+    if (!loggedInPerson) return;
+
+    try {
+      const fetchedAllocations = await allocationsApi.listAllocations({
+        startDate: new Date(),
+        endDate: new Date(),
+        personId: loggedInPerson?.id.toString()
+      });
+      const fetchedProjects = await projectsApi.listProjects({ startDate: new Date() });
+      const { filteredAllocations, filteredProjects } = filterAllocationsAndProjects(
+        fetchedAllocations,
+        fetchedProjects
+      );
+      const fetchedTimeEntries = await Promise.all(
+        filteredAllocations.map(async (allocation) => {
+          try {
+            if (allocation.project) {
+              const totalTimeEntries = await timeEntriesApi.listProjectTimeEntries({
+                projectId: allocation.project,
+                startDate: allocation.startDate,
+                endDate: allocation.endDate
+              });
+              let totalMinutes = 0;
+              totalTimeEntries.forEach((timeEntry: TimeEntries) => {
+                if (loggedInPerson && timeEntry.person === loggedInPerson.id) {
+                  totalMinutes += timeEntry.timeRegistered || 0;
+                }
+              });
+              return totalMinutes;
+            }
+          } catch (error) {
+            if (allocation.id) {
+              const message: string = strings
+                .formatString(
+                  strings.sprintRequestError.fetchAllocationError,
+                  allocation.id.toString(),
+                  error as string
+                )
+                .toString();
+              setError(message);
+            }
+          }
+          return 0;
+        })
+      );
+      setProjects(filteredProjects);
+      setAllocations(filteredAllocations);
+      setTimeEntries(fetchedTimeEntries);
+    } catch (error) {
+      setError(`${strings.sprintRequestError.fetchError}, ${error}`);
+    }
+    setLoading(false);
+  };
+
+  /**
+   * Calculate total unallocated time for the user in the current 2 week period
+   *
+   * @param allocation task allocated within a project
+   */
+  const unallocatedTime = (allocation: Allocations[]) => {
```

Comment:
Could this be called allocations as we are dealing with a list of allocations?

Reply:
This function calculates the remaining unallocated time for user. Calculates all expected for a sprint period and reduce it by the number of allocated hours, so I think unallocatedTime better describes functionality.

Reply:
Sorry, I wasn't clear, I was referring to the parameter name